adaptdl/adaptdl/torch/data.py

```diff
@@ -526,19 +490,19 @@ def __iter__(self):
             while not done:
                 self.sampler.set_epoch(
                     epoch, index=self._elastic.current_index)
-                self.batch_sampler.batch_size = self._elastic._sync_local_bsz()
+                self.batch_sampler.batch_size = self._elastic._context.get_batch_size()
```

Comment:
`_sync_local_bsz` cannot be replaced by the `Context.get_batch_size`, because `_sync_local_bsz` also does a broadcast from rank 0 to propagate the local batch size it calculated to rest of the replicas and also acts as barrier. `Context.get_batch_size` could cause local batch sizes in replicas to go out of sync.

Reply:
> `_sync_local_bsz` cannot be replaced by the `Context.get_batch_size`, because `_sync_local_bsz` also does a broadcast from rank 0 to propagate the local batch size it calculated to rest of the replicas and also acts as barrier. `Context.get_batch_size` could cause local batch sizes in replicas to go out of sync.

Thanks Omkar, please have a check on the new commit where the replacement issue is fixed.